lib/types.nix

```diff
@@ -352,6 +352,25 @@ rec {
       functor = (defaultFunctor name) // { wrapped = [ t1 t2 ]; };
     };
 
+    coercedTo = coercedType: coerceFunc: finalType: mkOptionType rec {
+      name = "coercedTo";
+      description = "${finalType.description} or ${coercedType.description} coerced to it";
+      check = x: coercedType.check x || finalType.check x;
```

Comment:
This does not matter much here, but I would swap the 2 conditions. (see why below)